### PR TITLE
[ui-build-to-maven] UI build process moved to maven

### PR DESCRIPTION
**Pull request by kpovetkin**

### CODE REVIEW

File: coopr-standalone/pom.xml [hunk 1]
```diff
@@ -35,6 +35,7 @@
         <module>../coopr-server</module>
         <module>../coopr-rest-client</module>
         <module>../coopr-cli</module>
+	<module>../coopr-ui</module>
```
Comment (nitinmotgi) [inline coopr-standalone/pom.xml]: ident correctly to above lines.

File: pom.xml [hunk 2]
```diff
@@ -31,6 +31,7 @@
         <module>coopr-server</module>
         <module>coopr-rest-client</module>
         <module>coopr-cli</module>
+	<module>coopr-ui</module>
```
Comment (nitinmotgi) [inline pom.xml]: same here

**Comment by kpovetkin** [comment]
@nitinmotgi Corrected

File: coopr-ui/build-ui.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+COOPR_NODE=`whereis node | awk '{print $2}'`
+COOPR_NPM=`whereis npm | awk '{print $2}'`
+
+rm -rf dist
```
Comment (gaarf) [inline coopr-ui/build-ui.sh]: there's a gulp task for that. `gulp clean`

File: coopr-ui/build-ui.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+${COOPR_NPM} run build && gulp distribute && ${COOPR_NPM} install --production
```
Comment (gaarf) [inline coopr-ui/build-ui.sh]: there shouldn't be a need for the final `&& ${COOPR_NPM} install --production`....

```
COOPR_GULP = './node_modules/gulp/bin/gulp.js'

${COOPR_NPM} run prebuild
${COOPR_NODE} ${COOPR_GULP} clean
${COOPR_NODE} ${COOPR_GULP} distribute
```

File: coopr-ui/build-ui.sh [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+COOPR_NODE=`whereis node | awk '{print $2}'`
+COOPR_NPM=`whereis npm | awk '{print $2}'`
```
Comment (wolf31o2) [inline coopr-ui/build-ui.sh]: Doesn't this just search the `PATH`, anyway? Seems redundant.

Comment (wolf31o2) [inline coopr-ui/build-ui.sh]: Also, these should be overridable by the user... Preferred would be:

```
COOPR_NODE=${COOPR_NODE:-node}
COOPR_NPM=${COOPR_NPM:-npm}
```

Comment (wolf31o2) [inline coopr-ui/build-ui.sh]: Looks like `whereis` isn't very reliable on every platform. On my Mac:

```
$ whereis node
$ which node
/usr/local/bin/node
```

Comment (anew) [inline coopr-ui/build-ui.sh]: haha. On my Mac:

```
$ whereis node
/usr/bin/node
$ which node
/usr/local/bin/node
```

Comment (gaarf) [inline coopr-ui/build-ui.sh]: on mine:

```
gaarf@ADs-MBP:~ $ whereis node
gaarf@ADs-MBP:~ $ which node
/usr/local/bin/node
gaarf@ADs-MBP:~ $ whereis bower
gaarf@ADs-MBP:~ $ which bower
/usr/local/bin/bower
gaarf@ADs-MBP:~ $ whereis java
/usr/bin/java
gaarf@ADs-MBP:~ $ which java
/System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home/bin/java
```

**Comment by kpovetkin** [comment]
Updated according comments

**Comment by wolf31o2** [comment]
Thanks!

**Pull request merged**